src/io/flutter/logging/tree/DataPanel.java

```diff
@@ -273,12 +276,8 @@ private JEditorPane createEditorPane() {
     return editorPane;
   }
 
-  private static ArrayList<DiagnosticsNode> getChildren(DiagnosticsNode node) {
+  private static List<DiagnosticsNode> getChildren(DiagnosticsNode node) {
     final ArrayList<DiagnosticsNode> children = node.getChildren().getNow(new ArrayList<>());
-    if (!children.isEmpty()) {
-      return children;
-    }
-
-    return node.getInlineProperties();
+    return Stream.of(children, node.getInlineProperties()).flatMap(Collection::stream).collect(Collectors.toList());
```

Comment:
nit: is this really easier to read then just adding the inline properties and children to an ArrayList?
I'm at a loss to why there is a flagMap call in the middle of this.

Reply:
Yeah.  Idiomatic Java is hard to read.  I'll sweep this on the next pas..